coordinator/points_writer.go

```diff
@@ -328,35 +315,18 @@ func (w *PointsWriter) WritePointsPrivileged(writeCtx tsdb.WriteContext, databas
 		}(writeCtx, shardMappings.Shards[shardID], database, retentionPolicy, points)
 	}
 
-	// Send points to subscriptions if possible.
-	var ok, dropped int64
-	pts := &WritePointsRequest{Database: database, RetentionPolicy: retentionPolicy, Points: points}
-	// We need to lock just in case the channel is about to be nil'ed
-	w.mu.RLock()
-	for _, ch := range w.subPoints {
-		select {
-		case ch <- pts:
-			ok++
-		default:
-			dropped++
-		}
-	}
-	w.mu.RUnlock()
-
-	if ok > 0 {
-		atomic.AddInt64(&w.stats.SubWriteOK, ok)
-	}
+	timeout := time.NewTimer(w.WriteTimeout)
+	defer timeout.Stop()
 
-	if dropped > 0 {
-		atomic.AddInt64(&w.stats.SubWriteDrop, dropped)
-	}
+	// Send points to subscriptions
+	pts := &WritePointsRequest{Database: database, RetentionPolicy: retentionPolicy, Points: points}
+	w.Subscriber.Send(pts)
```

Comment:
Is this guaranteed to eventually send the points, and that's why the "dropped" stat is gone?

Reply:
It might not send the points if the queues are full.

Before, we had one queue Input->Sorting, then sorting according to which subscription the points went to, then a bunch of Sorting->Sub1, Sorting->Sub2, Sorting->Sub3 ... .

The `SubWriteDrop` here was for if the Input->Sorting queue was full. Now we are guaranteed to find the right Sorting->Sub queue to put the write on, and either drop it or enqueue it, during `w.Subscriber.Send`. So we don't need the `SubWriteDrop` statistic here. We can still drop the write later, there is a different drop statistic later in the subscriber code for that.